Do not display password for self-hosted sites

malinajirka opened the pull request:
Fixes #9899 

This PR prevents an attacker from viewing a password for self-hosted sites.

![update-password](https://user-images.githubusercontent.com/2261188/58640810-25419280-82fa-11e9-8297-e5f15f216922.gif)


To test:
- Make sure the self-hosted site password is not visible ->My Sites > Settings > Tap on Password -> Notice the Edit text is empty
- Make sure that you can't set the password to an empty string
- Make sure a toast is displayed when you update the password
- Turn on airplane mode and notice all the preferences get disabled
- Make sure the password actually gets updated -> change it to a wrong password and go to My Site -> Blog Posts -> notice an error message in the LogCat

Update release notes:

- The change is too minor

**Note: Needs copy review -> the strings aren't final.**

michelleweber commented:
In the pop-up, "Update a password" seems like it should be "Enter a new password"? 

I'm a little confused about what the GIF is showing -- the screen looks like a normal log-in, the pop-up looks like a password change -- so I'm not sure if the copy is working.

loremattei commented:
Hey! I'm moving this to 12.7 since 12.6 has been cut. If you want it to land 12.6, please feel free to move it back, target the release branch and ping me to build a new beta.

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java:
```diff
@@ -1197,7 +1201,6 @@ public void setPreferencesFromSiteSettings() {
         changeEditTextPreferenceValue(mTaglinePref, mSiteSettings.getTagline());
         changeEditTextPreferenceValue(mAddressPref, mSiteSettings.getAddress());
         changeEditTextPreferenceValue(mUsernamePref, mSiteSettings.getUsername());
-        changeEditTextPreferenceValue(mPasswordPref, mSiteSettings.getPassword());
```
I felt like revealing the length of the password could help the attacker as well. But it's probably even more confusing for the user. Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9963/commits/bc44196daad096238f2ada08d76114677b8a2e18 and https://github.com/wordpress-mobile/WordPress-Android/pull/9963/commits/81678f04a06b930773bac918b0f1baa9b04ca8e6

malinajirka commented:
Thanks @theck13 for the great review!! I've committed all your proposed changes + fixed the issues you've found.

>I noticed the message when enabling airplane mode is "Logged out, editing disabled." It was a little misleading to me. What about changing that to "No connection. Editing is disabled." instead?

Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9963/commits/9ae874fc9397ca724a58cd5f5b2604b38f749cb4

>Also, the message duration is short. Should we update that to Duration.LONG?  

Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9963/commits/5cc3da91ba5cedb5769319df402cc207956fef11

>In the pop-up, "Update a password" seems like it should be "Enter a new password"?
I'm a little confused about what the GIF is showing -- the screen looks like a normal log-in, the pop-up looks like a password change -- so I'm not sure if the copy is working.

@michelleweber This is a settings screen for a self-hosted sites. Users can access it only after they successfully connect their site with the app. However, when they change their password using for example wp-admin, it breaks the connection with the app. This screen allows them to update the password the application uses to connect to their site - it won't change the site's password, it just updates the password the app uses for the connection. Does it make sense?

michelleweber commented:
Makes sense! In that case, I might suggest changing the headline to "Update password" and the text to something like "To reconnect the app to your self-hosted site, enter the site's new password here."

malinajirka commented:
Thanks Michelle! I've updated the texts. It's ready @theck13 - thanks!

![Screenshot_1559632091](https://user-images.githubusercontent.com/2261188/58858976-17e52900-86a9-11e9-8e30-8d1e9ffb9cec.png)

theck13 commented on WordPress/src/main/res/values/strings.xml:
```diff
@@ -540,7 +540,7 @@
     <string name="site_settings_language_title" translatable="false">@string/language</string>
     <string name="site_settings_username_title" translatable="false">@string/username</string>
-    <string name="site_settings_password_title" translatable="false">@string/password</string>
+    <string name="site_settings_password_title">Update password</string>
```
Changing the `site_settings_password_title` resource value changes not only the dialog, but also the preference title.  Was that intentional?  That makes it a little inconsistent with the ***Username*** preference.  Since long-pressing those preferences provides a hint for the action, should we leave the preferences ***Username*** and ***Password***?  @michelleweber, to keep the fields consistent, would it make sense to have just ***Password*** as the dialog title since there is a description of what the action does in the dialog message?

michelleweber replied:
I like the added clarity of having the added word in the title, since this is a very particular kind of password action, but if it will eff with other things to add it, leave it off. It's not that big a whoop.

nbradbury commented:
Just a friendly reminder to please make sure to manually merge `develop` into this branch before you merge this PR, as `develop` now contains the AndroidX migration. The automatic merge can succeed but still break the build. Better to find out now before it breaks `develop`. Thanks!

malinajirka commented:
Thanks @theck13: ;).

 >Changing the site_settings_password_title resource value changes not only the dialog, but also the preference title. Was that intentional?

Yes, it was intentional as I thought that's what we wanted. But you are right, it makes the preference inconsistent with the Username preference. I've fixed it in https://github.com/wordpress-mobile/WordPress-Android/pull/9963/commits/45d0075fa3bdf8026d69ad08b494943b6436600e. We show "Update password" in the dialog, but the preference row still uses just "Password".

I've also merged develop into the branch and verified everything works even with AndroidX;).

It's ready for another round. Thx
